[V2] Hide webpack performance warnings

vmarta opened the pull request:
# Description

Ticket: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000mQtDRYA0/view

This PR hides the Webpack's performance warnings for the `pwa` package, because we already have a similar `bundlesize` check in `pwa`.

When running `npm run build`:

Before:
![NeoVim Terminal 2022-03-30 at 17 06 02](https://user-images.githubusercontent.com/847300/160950544-4381b5fd-2634-4dc8-a999-c1c24a756014.png)

After: 
![NeoVim Terminal 2022-03-30 at 17 04 16](https://user-images.githubusercontent.com/847300/160950557-93da01f1-f42c-4678-88c7-121f4e4e2e7c.png)

And now running `npm run max-file-size` is also passing:
```bash
npm run test:max-file-size

> scaffold-pwa@2.0.0-dev.2 test:max-file-size /Users/vmarta/repos/pwa-kit/packages/pwa
> bundlesize


─ build/main.js
  ✔ build/main.js      47.73KB < 52 kB gzip

─ build/vendor.js
  ✔ build/vendor.js    240.02KB < 245 kB gzip

  2 checks passed
```

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [x] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# How to Test-Drive This PR
1. Switch to 'packages/pwa' directory
3. `npm run build`
5. Verify that the webpack performance hints are no longer showing


# Checklists

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [x] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

vmarta committed:
Hide webpack performance warnings

vmarta committed:
Update comment

adamraya commented on packages/pwa-kit-build/src/configs/webpack/config.js:
```diff
@@ -225,7 +225,11 @@ const client =
-                plugins: [...config.plugins, new LoadablePlugin({writeToDisk: true})]
+                plugins: [...config.plugins, new LoadablePlugin({writeToDisk: true})],
+                // Hide the performance hints, since we already have a similar `bundlesize` check in `pwa` package
+                performance: {
+                    hints: false
+                }
```
I know that hiding the warnings is in the ticket's AC, but Is it the right solution?

Instead of completely hiding the warning, I would suggest investigating why the bundle size increased, and if it makes sense for the size increase, update the file sizes here:
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/eefc03f4f6a465b2946c839a96e7e39c415047ed/packages/pwa-kit-build/src/configs/webpack/config.js#L209-L212

For context these are some of the findings that we got during the hackathon:
https://salesforce.quip.com/c2izAbgh3ch5#temp:C:TcMc89fe86c77c44d11bde45f0df

vmarta replied:
Good question, Adam. I didn't realize too that we set the performance limits in webpack. 

So I took a look at the bundle size increase, and it does make sense to me:
- First of all, with the bundle analyzer, I don't see anything suspicious or significant
- Then I saw in our Slack history that during the hackathon, Oli did not enable the minify option, so the bundle size became huge. But after enabling it, the numbers became much smaller and they looked to be all right with him (even though they're slightly above the limit)

vmarta replied:
I thought to myself, "ok let's update those numbers in the webpack config". But then I became unsure:
- Now we have 2 places where we check the bundle size: in webpack and also in `pwa`'s npm script. It's better to have just a single place to do it.
- Who's the target audience for the file size limit in webpack? It looks to be for ourselves only. Users have no way to update/configure the limit numbers because they're located in a different package (inside `pwa-kit-build`). OR they need to override it from their own webpack config, which may not be straightforward to do.

So IMO my suggestion is to do the following:
- Remove the file size limits from the webpack config
- Still hide Webpack's performance warnings (because by default, it's still checking the bundle size)
- And then update the size limit in pwa's package.json

What do you think? @adamraya

adamraya replied:
You're right @vmarta, it's not easy for users to update the bundle size in the Webpack config, so I agree that it's best to hide the warnings there.

I verified that currently, V2 bundles are slightly smaller than V1 bundles. 👍

One thing I noticed though, is that currently `npm run test:max-file-size` is falling on V2.


```
$ npm run test:max-file-size

> scaffold-pwa@2.0.0-dev.2 test:max-file-size /Users/arayanavarro/git/v2-pwa-kit/packages/pwa
> bundlesize


─ build/main.js
  ✖ build/main.js      47.73KB > 40 kB gzip

─ build/vendor.js
  ✔ build/vendor.js    239.94KB < 250 kB gzip

  1 check passed,  1 check failed
  
  ```
  
Still, I can run the `build`, `push` etc. commands without seeing any warnings, whereas the Webpack warnings are visible every time we build or push a bundle. Is that expected?
  
How are we communicating to users that the bundle size is bigger than the value set in `package.json`?
Should we run the `test:max-file-size` script when users `build` or `push` a project?

vmarta replied:
Yeah, currently we're not communicating to the users about their bundle size. Historically (as far as I can see), the `max-file-size` script was _not_ run together with other npm scripts like `prod:build` etc. I can see how it can be convenient for the users to be told that their bundle size has grown past the limit. On the other hand, it can be a source of friction too for them: since `bundlesize` outputs an error, that may prevent them from pushing a bundle.

Right now I'm thinking to stick with how things used to be. Leave `max-file-size` as a tool that users can choose to use in their workflow, but we're not forcing it on them.

vmarta replied:
Speaking of `max-file-size`: we used to be running it in our CI. But at some point, it's removed for all environments in [this PR](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/92#issuecomment-930497570). But I think we should bring it back such that it runs in non-Windows only.

adamraya replied:
Sounds good @vmarta. Let's leave it as it is. We can always enable warnings later if users start to reach MRT limits too often.

vmarta replied:
Ok, thanks @adamraya . FYI I've made the changes to webpack. Please review whenever you have time, thanks.

vmarta committed:
Update the `bundlesize` limits

And make sure that the pwa package bundles are checked in only 1 place (via `bundlesize`).

adamraya reviewed (approved):
LGTM

vmarta committed:
Merge branch 'v2' into hide-webpack-perf-warnings
